plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/Xenserver625StorageProcessor.java

```diff
@@ -100,36 +100,25 @@ protected SR createFileSR(final Connection conn, final String path) {
         PBD pbd = null;
 
         try {
-            final String srname = hypervisorResource.getHost().getUuid() + path.trim();
-
-            final Set<SR> srs = SR.getByNameLabel(conn, srname);
-
-            if (srs != null && !srs.isEmpty()) {
-                return srs.iterator().next();
+            final String srname = path.trim();
+            synchronized (srname.intern()) {
```

Comment:
All the calls touching external resource in the synchronized block are protected by a timeout. I think it will be redundant to add another layer of timeout on top of it. Even in case of this additional timeout there is no guarantee that the acquiring thread will be successful but will fail in all probability.